influxdb3_catalog/src/catalog.rs

```diff
@@ -414,14 +428,12 @@ impl InnerCatalog {
     }
 }
 
-#[serde_with::serde_as]
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
+#[derive(Debug, Eq, PartialEq, Clone)]
 pub struct DatabaseSchema {
     pub id: DbId,
     pub name: Arc<str>,
     /// The database is a map of tables
-    pub tables: SerdeVecHashMap<TableId, TableDefinition>,
-    #[serde_as(as = "TableMapAsArray")]
+    pub tables: SerdeVecMap<TableId, Arc<TableDefinition>>,
```

Comment:
Nice that the table definition is arc'd now. I'm guess that this makes it cheaper for the db schema to be updated when just a single table definition is updated as all the other tables will just be arc clones from the previous schema?

Reply:
Yeah, this is a nice side effect of `Arc`'ing it that I didn't really think of. I mainly did this to avoid having to use `&TableDefinition` everywhere and dealing with lifetime woes.